src/sknnr_spatial/estimator.py

```diff
@@ -0,0 +1,221 @@
+
+        self._wrapped_meta = FittedMetadata(
+            n_targets=self._get_n_targets(y),
+            target_names=self._get_target_names(y),
+        )
+
+        return self
+
+    @check_wrapper_implements
+    @check_is_x_image
+    def predict(
+        self, X_image: ImageType, *, nodata_vals: NoDataType = None
+    ) -> ImageType:
+        """
+        Predict target(s) for X_image.
+
+        Notes
+        -----
+        If X_image is not an image, the estimator's unmodified predict method will be
+        called instead.
+
+        Parameters
+        ----------
+        X_image : Numpy or Xarray image with 3 dimensions (y, x, band)
+            The input image. Features in the band dimension should correspond with the
+            features used to fit the estimator.
+        nodata_vals : float or sequence of floats, optional
+            NoData values to mask in the output image. A single value will be broadcast
+            to all bands while sequences of values will be assigned band-wise. If None,
+            values will be inferred if possible based on image metadata.
+
+        Returns
+        -------
+        y_image : Numpy or Xarray image with 3 dimensions (y, x, targets)
+            The predicted values.
+        """
+        wrapper = get_image_wrapper(X_image)(X_image, nodata_vals=nodata_vals)
+        return wrapper.predict(estimator=self)
+
+    @check_wrapper_implements
+    @check_is_x_image
+    def kneighbors(
+        self, X_image: ImageType, *, nodata_vals: NoDataType = None, **kneighbors_kwargs
+    ) -> ImageType | tuple[ImageType, ImageType]:
+        """
+        Find the K-neighbors of each pixel in an image.
+
+        Returns indices of and distances to the neighbors for each pixel.
+
+        Notes
+        -----
+        If X_image is not an image, the estimator's unmodified kneighbors method will be
+        called instead.
+
+        Parameters
+        ----------
+        X_image : Numpy or Xarray image with 3 dimensions (y, x, band)
+            The input image. Features in the band dimension should correspond with the
+            features used to fit the estimator.
+        nodata_vals : float or sequence of floats, optional
+            NoData values to mask in the output image. A single value will be broadcast
+            to all bands while sequences of values will be assigned band-wise. If None,
+            values will be inferred if possible based on image metadata.
+        **kneighbors_kwargs
+            Additional arguments passed to the estimator's kneighbors method, e.g.
+            `return_distance`.
+
+        Returns
+        -------
+        neigh_dist : Numpy or Xarray image with 3 dimensions (y, x, neighbor)
+            Array representing the lengths to points, only present if
+            return_distance=True.
+        neigh_ind : Numpy or Xarray image with 3 dimensions (y, x, neighbor)
+            Indices of the nearest points in the population matrix.
```

Comment:
I was leaning away from including anything `sknnr`-specific, but if you think removing the names on the returns would make it more general I'm all for that. I don't usually use names on my returns and only did here because I lazily adapted these from the sklearn docstrings, so I'm not even sure the correct practice for named returns - maybe they're supposed to match variables names within the function?

Reply:
> I assume this can return the index if called with return_dataframe_index=True? Totally a nit, but in this case we may not want to name the (unnamed) return value `neigh_ind` or perhaps add info to the description for this case?

I think you're absolutely correct here.  You have a good balance between being inclusive of `sklearn` (by including `return_distances`) and exclusive of `sknnr` (by leaving out `return_dataframe_index`).  I also think it's helpful to name the return values and the names you've chosen are perfectly appropriate.